File: projects/antia_numerical_methods@Beliavsky/c/fun/funk.c
/*	Function to calculate the integrand for calculating
	PSI(I,X) as required by function FREDCO.

	Function FKER(X,T) is the kernel K(x,t) and PHI(I,T) is the Ith
	basis function, phi_i(t). The argument X and I are passed through
	global variables XFRED and IFRED respectively.

	Functions FKER(X,T) and PHI(I,T) must be supplied by the user

	Required functions : FKER, PHI
*/

/*	To pass parameters from function FREDCO put these statements before fredco */
double XFRED;
int IFRED;

double funk(double t)

{
	return fker(XFRED,t)*phi(IFRED,t);
}

File: projects/antia_numerical_methods@Beliavsky/f77/sub8/funk.f
!	Function routine to calculate the integrand for calculating
!	PSI(I,X) as required by subroutine FREDCO.
!
!	Function FKER(X,T) is the kernel K(x,t) and PHI(I,T) is the Ith
!	basis function, phi_i(t). The argument X and I are passed through
!	the common block.
!
!	FUNCTION FKER(X,T) and FUNCTION PHI(I,T) must be supplied by the user
!
!	Required routines : FKER, PHI

      FUNCTION FUNK(T)
      IMPLICIT REAL*8(A-H,O-Z)
!	To pass parameters from subroutine FREDCO
      COMMON/ZZFRED/X,I

      FUNK=FKER(X,T)*PHI(I,T)
      END
